src/main/java/com/mihir/pipposample/Employee/EmployeeDao.java

```diff
@@ -0,0 +1,180 @@
+package com.mihir.pipposample.Employee;
+
+import com.google.inject.Inject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientOptions;
+import com.mongodb.MongoException;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.event.ServerHeartbeatFailedEvent;
+import com.mongodb.event.ServerHeartbeatStartedEvent;
+import com.mongodb.event.ServerHeartbeatSucceededEvent;
+import com.mongodb.event.ServerMonitorListener;
+import org.bson.Document;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class EmployeeDao implements ServerMonitorListener {
+    private enum HEALTH_STATUS{
+        STARTED, SUCCEEDED, FAILED
+    }
+
+    private HEALTH_STATUS status;
+    private MongoClient mongo;
+    private String mongoHost = "127.0.0.1";
+    private Integer mongoPort = 27017;
+    private String database = "EmployeeDB";
+    private String empCollection = "Employee";
+
+    Object healthCheckLock = new Object();
+
+    private ServerAddress servers;
+
+    @Inject
+    public EmployeeDao(){
+        super();
+
+        servers = new ServerAddress(mongoHost, mongoPort);
+        MongoClientOptions clientOptions = MongoClientOptions.builder()
```

Comment:
Inject a mongo client
Write a mongo client provider, guice

Reply:
like this https://github.com/unbxd/banshee/blob/6c3f5224634cb7ffb72a63a4f86010b243d7ab5f/commons/src/main/java/com/unbxd/commons/guice/providers/OkHttpClientProvider.java#L34-L33